provider/aci.go

```diff
@@ -1639,38 +1642,39 @@ func getProbe(probe *v1.Probe, ports []v1.ContainerPort) (*aci.ContainerProbe, e
 }
 
 func (p *ACIProvider) getAzureFileCSI(volume v1.Volume, namespace string) (*aci.Volume, error) {
-	azureSource := &aci.AzureFileVolume{
-		ReadOnly: *volume.CSI.ReadOnly,
-	}
-	if volume.CSI.NodePublishSecretRef != nil && volume.CSI.NodePublishSecretRef.Name != "" {
+	azureSource := &aci.AzureFileVolume{}
 
-		// Set the storage account name and key
-		secret, err := p.resourceManager.GetSecret(volume.CSI.NodePublishSecretRef.Name, namespace)
-		if err != nil {
-			return nil, errors.Wrap(err, fmt.Sprintf("The secret %s for AzureFile CSI driver %s is not found", volume.CSI.NodePublishSecretRef.Name, volume.Name))
-		}
+	if volume.CSI.VolumeAttributes != nil && len(volume.CSI.VolumeAttributes) != 0 {
```

Comment:
I would suggest write like this, this would check field name first, also add a shareName filed check.
```
var secretName, shareName string
if volume.CSI.VolumeAttributes != nil {
	for k, v := range volume.CSI.VolumeAttributes {
		switch strings.ToLower(k) {
		case azureFileSecretName:
			secretName := v
		case azureFileShareName:
			shareName = v
	}
}

if shareName == "" {
	return error ...
}
azureSource.ShareName = shareName

if secretName == "" {
	return error ...
}

secret, err := p.resourceManager.GetSecret(secretName, namespace)
if err != nil {
	return nil, errors.Wrap(err, fmt.Sprintf("the secret %s for AzureFile CSI driver %s is not found", secretName, volume.Name))
}
if secret == nil {
	return nil, fmt.Errorf("getting secret for AzureFile CSI driver %s returned an empty secret", volume.Name)
}

// Set the storage account name and key
azureSource.StorageAccountName = strings.TrimSpace(string(secret.Data[azureFileStorageAccountName]))
azureSource.StorageAccountKey = strings.TrimSpace(string(secret.Data[azureFileStorageAccountKey]))
```